### PR TITLE
configure smt and indep_thread_mode for Power8 and Power9 (compat mode) 

### PR DESCRIPTION
**Pull request by balamuruhans**
For Power8 and Power9 (compat mode) host smt have to be turned off and indep_thread_mode have to
be configured as 'N' for guest to boot successfully. This configuration have to be considered for local host
and for remote host (In case of migration).

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: shared/cfg/machines.cfg [hunk 1]
```diff
@@ -30,12 +30,18 @@ variants:
         machine_type = pseries
         # Currently only supports standard VGA
         # Make sure below param is set 'yes' to run a
-        # power8(compat mode) on power9 host
-        # power9_compat = "no"
+        # power8 on power9(compat mode) host
```
Comment (sathnaga) [inline shared/cfg/machines.cfg]: s/` power8 on power9(compat mode) host`/`power8(compat mode guest) on power9 host`

File: virttest/test_setup.py [hunk 2]
```diff
@@ -2042,6 +2042,76 @@ def stop(self):
...
+    server_ip = params["server_ip"]
+    server_user = params.get("server_user", "root")
+    server_pwd = params["server_pwd"]
+    return remote.wait_for_login('ssh', server_ip, '22', server_user,
+                                 server_pwd, r"[\#\$]\s*$")
+
+
+def switch_indep_threads_mode(state="Y", params=None):
+    """
+    For POWER8 guest to boot in POWER9 compat mode, indep_threads_mode
```
Comment (sathnaga) [inline virttest/test_setup.py]: same here

File: virttest/env_process.py [hunk 3]
```diff
@@ -1083,15 +1094,24 @@ def postprocess(test, params, env):
...
+    power9_compat_remote = params.get("power9_compat_remote", "no") == "yes"
+    if restore_smt_remote or power9_compat_remote:
+        cmd = "grep cpu /proc/cpuinfo | awk '{print $3}' | head -n 1"
+        server_session = test_setup.remote_session(params)
+        cmd_output = server_session.cmd_status_output(cmd)
+        server_session.close()
+        if (cmd_output[0] == 0):
+            cmd_output = cmd_output[1].strip().lower()
+            if "power8" in cmd_output:
+                test_setup.switch_smt(state="on", params=params)
```
Comment (sathnaga) [inline virttest/env_process.py]: I don't think there is a usecase for smt=on in power8 kvm host, so we can skip power8 from restoring smt.

**Comment by balamuruhans** [comment]
@sathnaga made the changes as suggested.

File: virttest/test_setup.py [hunk 4]
```diff
@@ -2042,6 +2042,76 @@ def stop(self):
...
+            server_session.close()
+            raise exceptions.TestSetupFail("Unable to get indep_threads_mode:"
+                                           " %s" % cmd_output[1])
+        thread_mode = cmd_output[1].strip()
+    else:
+        try:
+            thread_mode = process.system_output(cmd, shell=True)
+        except process.CmdError as info:
+            thread_mode = info.result.stderr.strip()
+            raise exceptions.TestSetupFail("Unable to set indep_threads_mode "
```
Comment (sathnaga) [inline virttest/test_setup.py]: failure msg to be changed, `unable to get indep_threads_mode`

File: virttest/test_setup.py [hunk 5]
```diff
@@ -2042,6 +2042,76 @@ def stop(self):
...
+            if (server_user.lower() != "root"):
+                raise exceptions.TestSkipError("Turning indep_thread_mode %s "
+                                               "requires root privileges "
+                                               "(currently running "
+                                               "with user %s)" % (state,
+                                                                  server_user))
+            cmd_output = server_session.cmd_status_output(cmd)
+            server_session.close()
+            if (cmd_output[0] != 0):
+                raise exceptions.TestSetupFail("Unable to turn %s  :%s"
```
Comment (sathnaga) [inline virttest/test_setup.py]: expand the error `Unable to turn %s indep_thread_mode`

File: virttest/test_setup.py [hunk 6]
```diff
@@ -2042,6 +2042,76 @@ def stop(self):
...
+                logging.debug("indep_thread_mode turned %s successfully "
+                              "in remote server", state)
+        else:
+            try:
+                utils_misc.verify_running_as_root()
+                process.run(cmd, verbose=True, shell=True)
+                logging.debug("indep_thread_mode turned %s successfully",
+                              state)
+            except process.CmdError, info:
+                raise exceptions.TestSetupFail("Unable to turn %s SMT :%s" %
```
Comment (sathnaga) [inline virttest/test_setup.py]: Wrong error message...

**Comment by balamuruhans** [comment]
@sathnaga made the changes, please help to merge it.

**Pull request merged**